libminifi/include/utils/OptionalUtils.h

```diff
@@ -37,11 +39,47 @@ optional<typename gsl_lite::remove_cvref<T>::type> optional_from_ptr(T&& obj) {
   return obj == nullptr ? nullopt : optional<typename gsl_lite::remove_cvref<T>::type>{ std::forward<T>(obj) };
 }
 
+template<typename>
+struct is_optional : std::false_type {};
+
+template<typename T>
+struct is_optional<optional<T>> : std::true_type {};
+
 }  // namespace utils
 }  // namespace minifi
 }  // namespace nifi
 }  // namespace apache
 }  // namespace org
 
+namespace nonstd {
+namespace optional_lite {
+
+namespace minifiutils = org::apache::nifi::minifi::utils;
+
+// map
+template<typename SourceType, typename F>
+auto operator|(const optional<SourceType>& o, F&& f) noexcept(noexcept(minifiutils::invoke(std::forward<F>(f), *o)))
+    -> optional<typename std::decay<decltype(minifiutils::invoke(std::forward<F>(f), *o))>::type> {
+  if (o.has_value()) {
+    return make_optional(minifiutils::invoke(std::forward<F>(f), *o));
+  } else {
+    return nullopt;
+  }
+}
+
+// bind
+template<typename SourceType, typename F>
+auto operator>>=(const optional<SourceType>& o, F&& f) noexcept(noexcept(minifiutils::invoke(std::forward<F>(f), *o)))
```

Comment:
I'd rather we use the conventional `map` and `flatMap` names, could we create a custom `optional` class (with `map` and `flatMap`) extending the `nonstd::optional`, and make that available as `utils::optional`?

Reply:
I'd prefer to avoid modifying the optional class. What do you think about this syntax?
```
utils::optional<T> opt{...};
auto map_result = opt | utils::map([](T) -> U {...});
auto flat_map_result = opt | utils::flatMap([](T) -> utils::optional<U>{...});
```

Reply:
yes, that could work

Reply:
fixed in 13116f0 and fdcec88